languages/python/sqlalchemy-oso/sqlalchemy_oso/partial.py

```diff
@@ -44,7 +45,32 @@ def flip_op(operator):
 def partial_to_filter(expression: Expression, session: Session, model, get_model):
     """Convert constraints in ``partial`` to a filter over ``model`` that should be applied to query."""
     expression = preprocess(expression)
-    return translate_expr(expression, session, model, get_model)
+
+    role_allows = False
+    if contains_role_allows(expression):
+        role_allows = True
+
+    return (translate_expr(expression, session, model, get_model), role_allows)
+
+
+def contains_role_allows(expression: Expression):
+    assert expression.operator == "And"
+    for expr in expression.args:
+        op = expr.operator
+        left = expr.args[0]
+        right = expr.args[1]
+        if (
+            op == "Unify"
```

Comment:
We have to check for this bidirectionally (we can't rely on `True` being on the left and `Dot` on the right).